firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/UpdateTaskImpl.java

```diff
@@ -32,14 +32,17 @@
 import com.google.firebase.appdistribution.UpdateProgress;
 import com.google.firebase.appdistribution.UpdateTask;
 import com.google.firebase.concurrent.FirebaseExecutors;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.Executor;
 
 /** Implementation of UpdateTask, the return type of updateApp. */
 // TODO(b/261013814): Use an explicit executor in continuations.
 class UpdateTaskImpl extends UpdateTask {
+
   @Nullable
   @GuardedBy("lock")
-  private ManagedListener listener = null;
+  private List<ManagedListener> listeners = new ArrayList<>();
```

Comment:
The update to `updateApp_whenCalledMultipleTimesWithAAB_onlyMakesOneRequest` below exposed this bug - we weren't supporting multiple listeners for progress events.